05-UI-ReactJS/react-builder-master/source/Movie.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import MovieInput from './MovieInput';
+import MovieList from './MovieList';
+import MovieEdit from './MovieEdit';
+
+class Movie extends React.Component{
+  render () {
+    return (
+        <div className="movie">
+            <MovieInput onSubmit={this.handleMovieInputSubmit} /> <br />
+            <MovieList movies={this.getMovies()} />
+            <h3 id="favouriteListTitle" />
+            <ul id="list" />
+            <MovieEdit movies={this.getMovies()} onSubmit={this.handleUpdateMovie} />
+        </div>
+    );
+  }
+
+  getMovies () {
+    let movies = localStorage.getItem("movieStorage") || [];
+    if(movies != ''){
```

Comment:
Two notes here:
* Remember to evaluate value **and** type in your conditionals instead of != use *!==* , the first only evaluates the value of the variables whereas the second one evaluates value and type as well.
* Always try to have one *return* per function, you could refactor this to something like:
````
let result;
let movies = localStorage.getItem('movieStorage') || []:
if (movies.length) { //This evaluates that your array is not empty
    result = JSON.parse(movies);
} else {
    result = JSON.parse(null);
}

return result; //This way you only have one return in your function
````